game-core/src/test/java/games/strategy/triplea/UnitUtilsTest.java

```diff
@@ -0,0 +1,129 @@
+  private final UnitType battleship = GameDataTestUtil.battleship(gameData);
+  private final UnitType factory = GameDataTestUtil.factory(gameData);
+
+  @Nested
+  class TranslateAttributesToOtherUnits {
+
+    @Test
+    void noChangesIfNothingToTransfer() {
+      final Unit oldUnit = battleship.create(1, player).get(0);
+      final Collection<Unit> newUnits = battleship.create(1, player);
+
+      final Change changes = UnitUtils.translateAttributesToOtherUnits(oldUnit, newUnits, seaZone);
+
+      assertThat(changes.isEmpty(), is(true));
+    }
+
+    @Test
+    void hitsAreTransferred() {
+      final Unit oldUnit = battleship.create(1, player).get(0);
+      final List<Unit> newUnits = battleship.create(1, player);
+      oldUnit.setHits(1);
+
+      final Change changes = UnitUtils.translateAttributesToOtherUnits(oldUnit, newUnits, seaZone);
+      gameData.performChange(changes);
+
+      assertThat(newUnits.get(0).getHits(), is(1));
+    }
+
+    @Test
+    void hitsAreTransferredToAllUnits() {
+      final Unit oldUnit = battleship.create(1, player).get(0);
+      final List<Unit> newUnits = battleship.create(2, player);
+      oldUnit.setHits(1);
+
+      final Change changes = UnitUtils.translateAttributesToOtherUnits(oldUnit, newUnits, seaZone);
+      gameData.performChange(changes);
+
+      assertThat(newUnits.get(0).getHits(), is(1));
+      assertThat(newUnits.get(1).getHits(), is(1));
+    }
+
+    @Test
+    void newUnitHasAtLeast1HP() {
+      final Unit oldUnit = battleship.create(1, player).get(0);
+      final List<Unit> newUnits = battleship.create(1, player);
+      oldUnit.setHits(2);
+
+      final Change changes = UnitUtils.translateAttributesToOtherUnits(oldUnit, newUnits, seaZone);
+      gameData.performChange(changes);
+
+      assertThat(
+          "The new unit will always have at least 1 HP, even if the number of hits to transfer "
+              + "would have set it to 0 HP.",
+          newUnits.get(0).getHits(),
+          is(1));
+    }
+
+    @Test
+    void bombingDamageIsTransferred() {
+      final Unit oldUnit = factory.create(1, player).get(0);
+      final List<Unit> newUnits = factory.create(1, player);
+      oldUnit.setUnitDamage(5);
+
+      final Change changes = UnitUtils.translateAttributesToOtherUnits(oldUnit, newUnits, landZone);
+      gameData.performChange(changes);
+
+      assertThat(newUnits.get(0).getUnitDamage(), is(5));
+    }
+
+    @Test
+    void bombingDamageIsLimited() {
+      final Unit oldUnit = factory.create(1, player).get(0);
+      final List<Unit> newUnits = factory.create(1, player);
+      // force a really large amount of damage
+      oldUnit.setUnitDamage(100);
+
+      final Change changes = UnitUtils.translateAttributesToOtherUnits(oldUnit, newUnits, landZone);
+      gameData.performChange(changes);
+
+      assertThat(
+          "Factory can only have 20 damage in Germany", newUnits.get(0).getUnitDamage(), is(20));
```

Comment:
:+1: 
Like the comment on the assertion, I was wondering why the 100 was translated to 20, this comment answered that perfectly.